recipe/run_test.sh

```diff
@@ -50,6 +50,16 @@ unittest() {
   msg OK
 }
 
+notebooks() {
+  msg Running notebook tests
+  (
+    set -eux
+    cd ../notebooks
+    pytest ./tests
+  )
+  msg OK
+}
```

Comment:
Looking good. Here's the more design-oriented discussion, for everyone...

1. This file, `recipe/run_test.sh` technically belongs to conda: It's what `conda build` runs when it's building a package, to ensure that the code being packaged passes all tests before the final package is made available. To make sure that `uwtools` developers run (and pass) exactly the same tests that `conda build` will run -- so that we have high confidence that our CI pipeline will pass -- our top-level `Makefile` calls into `run_test.sh` to execute tests when we run `make test` or the more-granular `make lint`, `make unittest`, etc.

Since @Byrnetp wisely did not add `notebooks` to the full list of checks on lines 68-72, below, `conda build` should still work fine, because it will not run this `notebooks()` function. It would not pass if it run it, though, because the packages needed to run the notebook tests, especially `testbook` would not be installed -- see the `test.requires` section in the `meta.yaml` file adjacent to this file. So, all good so far.

But given that, I think we don't need this function at all: I think we could instead update @Byrnetp 's addition to the top-level `Makefile` to say something like

```
testbook:
    $(MAKE) -C notebooks test
```

(note how this is similar to the `docs` target; also, I'm just throwing out `testbook` as a shorter alternative to `test-notebooks`, but I'm ok either way) and creating `notebooks/Makefile` with
```
TARGETS = testbook

.PHONY: $(TARGETS)

all:
	$(error Valid targets are: $(TARGETS))

testbook:
	pytest tests
```
This would allow developers to run `make testbook` from either the repo root or, if they're already working there, from the `notebooks/` directory.

2. To run these tests, I found that I needed to install the `testbook` package into my development environment / shell. That initially failed with a complaint about `no such kernel named python3`. Probably installing `jupyterlab` would fix this, but that's quite a heavyweight package, and some searching suggested installing `nb_conda_kernels`, which may be more lightweight. Given this, I think we might need a `notebooks/install-deps` file similar, in spirit, to `docs/install-deps`, but probably simpler, like:
```
conda install -q -y nb_conda_kernels testbook
```
When we eventually add notebook-related documentation to our _Contributor Guide_ RST docs, we would mention that one needs to `source notebooks/install-deps` in a `uwtools` development environment before running e.g. `make testbook`. We mention this step for documentation contributors, for their `install-deps`. (Maybe later we'll want to pin the versions of those two packages, but we can probably ignore that for now.)

3. Finally, I'll say that I do think we'll eventually want to add steps to our CI configuration to run the notebook tests. Again, we have CI tests for the documentation, to test e.g. for broken symlinks, but those are not run from `run_test.sh` because they're not necessary for conda packaging, and I think the same is true for the notebook tests.

What does everybody think?

Reply:
I'll add that, when I install `testbook` and `nb_conda_kernels` and run the tests, I'm seeing a warning:
```
=============================================== warnings summary ===============================================
../../../sw/conda/envs/UPSTREAM-uwtools/lib/python3.12/site-packages/jupyter_client/connect.py:22
  /home/maddenp/sw/conda/envs/UPSTREAM-uwtools/lib/python3.12/site-packages/jupyter_client/connect.py:22: DeprecationWarning: Jupyter is migrating its paths to use standard platformdirs
  given by the platformdirs library.  To remove this warning and
  see the appropriate new directories, set the environment variable
  `JUPYTER_PLATFORM_DIRS=1` and then run `jupyter --paths`.
  The use of platformdirs will be the default in `jupyter_core` v6
    from jupyter_core.paths import jupyter_data_dir, jupyter_runtime_dir, secure_write

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================= 2 passed, 1 warning in 1.93s =========================================
```

I haven't looked into it, but would be interested to know if anyone else sees this. I also see it if I install `jupyterlab` instead of `nb_conda_kernels`.

It could be that the warnings are due to code in dependency packages that we can't do anything about. If so, there are ways to suppress specific warning messages, so we don't necessarily have to live with this.

Reply:
All of those suggestions sound good to me! Using `testbook` as the target name seems perhaps too specific or potentially confusing to me but I agree that it would be nice if it was shorter; what do you all think of using `test-nb` instead?

I was also getting that warning when running the tests, and I was setting the environment variable it mentions using `export JUPYTER_PLATFORM_DIRS=1` to silence it. I'm not sure what the best way to suppress it would be.

Reply:
Try creating `notebooks/pyproject.toml` with content
```
[tool.pytest.ini_options]
filterwarnings = ["ignore::DeprecationWarning"]
```
and see if the warning goes away for you -- it does for me. If that works, you could add/commit/push that.

`test-nb` works for me.

Reply:
This all sounds good to me. Thanks for making these updates, Travis.